### PR TITLE
ISOC-3692 add code scanning alerts backfill

### PR DESCRIPTION
**Pull request by auzwang**
**What's in this PR?**
Adds Security in Jira `code_scanning_alert` backfill support. GitHub Security [Code Scanning Alerts](https://docs.github.com/en/rest/code-scanning/code-scanning?apiVersion=2022-11-28#list-code-scanning-alerts-for-a-repository) will be transformed and submitted as [Security Vulnerabilities in Jira](https://developer.atlassian.com/cloud/jira/software/rest/api-group-security-information/#api-rest-security-1-0-linkedworkspaces-bulk-post).

Code scanning alerts from PRs are not included in the REST API by default.

**Why**
Integrate GitHub with Security in Jira
https://hello.atlassian.net/wiki/spaces/CDX/pages/2673460294/RFC+GitHub+for+Jira+Security

**Added feature flags**
ENABLE_GITHUB_SECURITY_IN_JIRA

**Affected issues**
[ISOC-3692](https://softwareteams.atlassian.net/browse/ISOC-3692)

**How has this been tested?**  
Locally
Unit Tests

**Todo**
- Do not merge until migrations have been run: https://github.com/atlassian/github-for-jira/pull/2362
- Scheduler test

[ISOC-3692]: https://softwareteams.atlassian.net/browse/ISOC-3692

### CODE REVIEW

**Comment by Harminder84** [comment]
Make a seperate PR for DB changes and run the DB script on all environments. Here is the run book to run DB schema migration - https://hello.atlassian.net/wiki/spaces/OTFS/pages/1958321242/HOWTO+Run+db+schema+migration+manually+on+ddev+stg+prod

File: src/sync/code-scanning-alerts.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+	const jiraPayload = await transformCodeScanningAlert(codeScanningAlerts, repository, jiraHost, logger,  messagePayload.gitHubAppConfig?.gitHubAppId);
+	logger.info({ processingTime: Date.now() - startTime, jiraPayloadLength: jiraPayload?.vulnerabilities?.length }, "Backfill task complete");
+	return {
+		edges: edgesWithCursor,
+		jiraPayload
+	};
+};
+
+const areAllBuildsEarlierThanFromDate = (alerts: CodeScanningAlertResponseItem[], fromDate: Date | undefined): boolean => {
```
Comment (Harminder84) [inline src/sync/code-scanning-alerts.ts:66]: This method name should be something like `areAllAlertsEarlierThanFromDate`

**Review by gxueatlassian — CHANGES_REQUESTED** [review]
LGTM, just blocking until DB migrated.

**Comment by auzwang** [comment]
@gxueatlassian the DB migrations have been successfully run on all environments. Thanks @Harminder84 !

**Review by gxueatlassian — APPROVED** [review]

**Pull request merged**